underfs/swift/src/main/java/alluxio/underfs/swift/SwiftUnderFileSystem.java

```diff
@@ -227,10 +239,13 @@ public boolean exists(String path) throws IOException {
       return true;
     }
 
-    // Query file or folder using single listing query
-    Collection<DirectoryOrObject> objects =
-        listInternal(stripFolderSuffixIfPresent(stripContainerPrefixIfPresent(path)), false);
-    return objects != null && objects.size() != 0;
```

Comment:
exists was returning true for 'swift://swift-container/dir' even if only 'swift://swift-container/dir1' existed because of the list call with prefix 'swift://swift-container/dir'.